iohub/ngff/nodes.py

```diff
@@ -1010,6 +1010,80 @@ def get_axis_index(self, axis_name: str) -> int:
         """
         return self.axis_names.index(axis_name.lower())
 
+    def get_effective_scale(
+        self,
+        image: str | Literal["*"],
+    ) -> TransformationMeta:
+        """Get the total coordinate scale metadata
+        for one image array or the whole FOV.
+
+        Parameters
+        ----------
+        image : str | Literal["*"]
+            Name of one image array (e.g. "0") to query,
+            or "*" for the whole FOV
+        """
+        transforms: list[TransformationMeta] = [
+            t for t in self.metadata.multiscales[0].coordinate_transformations
+        ]
+        if image != "*" and image in self:
+            for i, dataset_meta in enumerate(
+                self.metadata.multiscales[0].datasets
+            ):
+                if dataset_meta.path == image:
+                    transforms.extend(
+                        self.metadata.multiscales[0]
+                        .datasets[i]
+                        .coordinate_transformations
+                    )
+        elif image != "*":
+            raise ValueError(f"Key {image} not recognized.")
```

Comment:
This block is duplicated between the two methods. Can you abstract it out as a shared private method?